CppCoreGuidelines.md

```diff
@@ -6539,6 +6540,40 @@ That tends to work better than "cleverness" for non-specialists.
 
 * Flag throwing `hash`es.
 
+### <a name="Rc-memset"></a>C.90: Rely on constructors and assignment operators, not `memset` and `memcpy`
+
+##### Reason
+
+The standard C++ mechanism to construct a an instance of a type is to call its constructor. As specified in [C.41: A constructor should create a fully initialized object](#Rc-complete). No additional initialization, such as by `memcpy`, should be required.
+A type will provide a copy constructor and/or copy assignment operator to appropriately make a copy of the class, preserving the type's invariants.  Using memcpy to copy a non-trivially copyable type can result in slicing, or data corruption. 
```

Comment:
I think it would be better to categorically state it has undefined behaviour, and then say that can result in slicing or data corruption. The current wording could be read that sometimes it works OK, whereas it's always undefined.